frontend/integration-tests/views/secrets.view.ts

```diff
@@ -1,12 +1,54 @@
-import { $, $$ } from 'protractor';
+/* eslint-disable no-unused-vars */
+import { $, $$, browser, by, ExpectedConditions as until, element, ElementFinder } from 'protractor';
+
+import * as crudView from '../views/crud.view';
 
 export const secretNameInput = $('#secret-name');
 export const pre = $$('.co-copy-to-clipboard__text');
 
 export const saveButton = $('#save-changes');
 export const errorMessage = $('.alert-danger');
+export const authTypeDropdown = $('#dropdown-selectbox');
+
+export const webhookSecretValueInput = $('input[name=webhookSecretKey]');
+export const sourceSecretUsernameInput = $('input[name=username]');
+export const sourceSecretPasswordInput = $('input[name=password]');
+export const sourceSecretSSHTextArea = $('.co-file-dropzone__textarea');
 
 export const webhookSecretValue = 'webhookValue';
-export const webhookSecretValueInput = $('#webhook-secret-key');
+export const sourceSecretUsername = 'username';
+export const sourceSecretUsernameUpdated = 'usernameUpdated';
+export const sourceSecretPassword = 'password';
+export const sourceSecretPasswordUpdated = 'passwordUpdated';
+export const sourceSecretSSHKey = 'sshKey';
+export const sourceSecretSSHKeUpdated = 'sshKeyUpdated';
 
 export const createWebhookSecretLink = $('#webhook-link');
+export const createSourceSecretLink = $('#source-link');
+
+export const createSecret = async(linkElement: ElementFinder, name: string, func: Function) => {
+  await crudView.createItemButton.click();
+  await linkElement.click();
+  await secretNameInput.sendKeys(name);
+  await func();
+  await saveButton.click();
+  expect(errorMessage.isPresent()).toBe(false);
+};
+
+export const checkSecret = async(name: string, valuesToCheck: string[]) => {
+  await browser.wait(until.textToBePresentInElement($('.co-m-pane__heading'), name));
+  await element(by.partialButtonText('Reveal Values')).click();
+  valuesToCheck.forEach(async(value, index) => {
+    expect(pre.get(index).getText()).toEqual(value);
+  });
+};
+
+export const editSecret = async(ns: string, name: string, func: Function) => {
```

Comment:
I'd give `func` a more precise name like `updateForm` (here and in `createSecret`)